src/components/Libraries/LibrarySettingsPane.tsx

```diff
@@ -0,0 +1,281 @@
+
+  const [descValue, setDescValue] = useState(description);
+
+  const [isChecked, setIsChecked] = useState(isPublic);
+
+  const [isSaving, setIsSaving] = useState(false);
+
+  const modified = useMemo(
+    () => name !== nameValue || description !== descValue || isPublic !== isChecked,
+    [nameValue, descValue, name, description, isChecked, isPublic],
+  );
+
+  const canEdit = permission === 'owner' || permission === 'admin';
+
+  const handleOnNameChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setNameValue(e.target.value);
+  };
+
+  const handleOnDescChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    setDescValue(e.target.value);
+  };
+
+  const handleCheckboxChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setIsChecked(e.target.checked);
+  };
+
+  const handleSave = () => {
+    setIsSaving(true);
+    updateMeta(
+      {
+        id,
+        description: descValue !== description ? descValue : undefined,
+        name: nameValue !== name ? nameValue : undefined,
+        public: isPublic !== isChecked ? isChecked : undefined,
+      },
+      {
+        onSettled(data, error) {
+          if (error) {
+            toast({
+              status: 'error',
+              title: 'Error saving changes',
+              description: parseAPIError(error),
+            });
+          } else {
+            toast({
+              status: 'success',
+              title: 'Updated',
+            });
+            onRefetch();
+          }
+          setIsSaving(false);
+        },
+      },
+    );
+  };
+
+  const handleTransfer = (newUser: string) => {
+    transfer(
+      { id, email: newUser },
+      {
+        onSettled(data, error) {
+          if (error) {
+            toast({
+              status: 'error',
+              title: 'Error transfering ownership',
+              description: parseAPIError(error),
+            });
+          } else {
+            toast({
+              status: 'success',
+              title: `Successfully transferred ownership to ${newUser}`,
+            });
+            void router.push({ pathname: '/user/libraries' });
+          }
+        },
+      },
+    );
+  };
+
+  const handleDeleteLibrary = () => {
+    deleteLibrary(
+      { id },
+      {
+        onSettled(data, error) {
+          if (error) {
+            toast({
+              status: 'error',
+              title: 'Error deleting library',
+              description: parseAPIError(error),
+            });
+          } else {
+            toast({
+              status: 'success',
+              title: 'Library deleted',
+            });
+            void router.push({ pathname: '/user/libraries' });
+          }
+        },
+      },
+    );
+  };
+
+  return (
+    <>
+      <Box>
+        <Heading variant="pageTitle" as="h1" my={4}>
+          <SimpleLink href={`/user/libraries/${id}`} display="inline">
+            <ChevronLeftIcon mr={2} />
+          </SimpleLink>
```

Comment:
We should have a component for these back links.  This should work like a stack that we 'push' as we dig in, and 'pop' as we go back up

Reply:
What's the advantage for the component? I can add a task for this.

Reply:
Yeah we could think about it anyway, we should probably use hash urls or something like that as well for the sub-views so that we could keep track of it/shareable/etc.